File: go-client-sdk/pkg/models/operations/postinvoicesinvoiceidvoid.go
package operations

import (
	"github.com/speakeasy-sdks/orb-sdks/go-client-sdk/v2/pkg/models/shared"
)

type PostInvoicesInvoiceIDVoidPathParams struct {
	InvoiceID string `pathParam:"style=simple,explode=false,name=invoice_id"`
}

type PostInvoicesInvoiceIDVoidRequest struct {
	PathParams PostInvoicesInvoiceIDVoidPathParams
}

type PostInvoicesInvoiceIDVoidResponse struct {
	ContentType string
	Invoice     *shared.Invoice
	StatusCode  int
}

File: typescript-client-sdk/src/sdk/models/operations/postinvoicesinvoiceidvoid.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class PostInvoicesInvoiceIdVoidPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=invoice_id" })
  invoiceId: string;
}


export class PostInvoicesInvoiceIdVoidRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: PostInvoicesInvoiceIdVoidPathParams;
}


export class PostInvoicesInvoiceIdVoidResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  invoice?: shared.Invoice;

  @SpeakeasyMetadata()
  statusCode: number;
}
